pages/sensors/[id].tsx

```diff
@@ -1,28 +1,234 @@
+    endDateTimeString: string | undefined;
+  }>({
+    startDateTimeString: tenDaysAgo.toISOString(),
+    endDateTimeString: today.toISOString(),
+  });
+  const { lastRecordDate } = useSensorLastRecordDate(sensor.id);
+  const { count: recordsCount } = useSensorRecordsCount(sensor.id);
+  const {
+    records,
+    error: recordsFetchError,
+    isLoading: recordsAreLoading,
+  } = useSensorRecords({
+    sensorId: sensor.id,
+    startDateString: currentDatetimeRange.startDateTimeString,
+    endDateString: currentDatetimeRange.endDateTimeString,
+  });
+  const parsedAndSortedRecords = createDateValueArray(records);
+
+  useEffect(() => {
+    if (!lastRecordDate) return;
+    setCurrentDatetimeRange({
+      startDateTimeString: moment
+        .parseZone(lastRecordDate)
+        .subtract(7, "days")
+        .toISOString(),
+      endDateTimeString: moment.parseZone(lastRecordDate).toISOString(),
+    });
+  }, [lastRecordDate, setCurrentDatetimeRange]);
+
+  const chartWrapper = useCallback((node: HTMLDivElement | null) => {
+    if (!node) return;
+    const { width } = node.getBoundingClientRect();
+    setChartWidth(width || 0);
+    setChartHeight(width / 2);
+  }, []);
+
   return (
     <>
-      <h1>Sensor page: {sensor.name || ""}</h1>
-      <p>I am a sensor page</p>
-      <p>ID: {sensor.id}</p>
+      <SensorPageHeader
+        id={sensor.id}
+        name={sensor.name || ""}
+        description={sensor.description}
+        category={sensor.category}
+        symbol={sensor.icon_id || fallbackIconId.current}
+        geocoordinates={{ latitude: 52.4961458, longitude: 13.4335723 }}
+        author={{
+          username: sensor.user.name || "anonymous",
+          displayName: sensor.user.display_name || "Anonymous",
+        }}
+        withEditButton={!!user && user.name === sensor.user?.name}
+      />
+      <div className='container mx-auto max-w-8xl mb-32'>
+        <div>
+          <div className='flex justify-between flex-wrap gap-4 pb-8 px-4'>
+            <DeviceLineChartFilters
+              startDateTimeString={currentDatetimeRange.startDateTimeString}
+              endDateTimeString={currentDatetimeRange.endDateTimeString}
+              onDatetimeRangeChange={vals => setCurrentDatetimeRange(vals)}
+            />
+            <div className='md:pt-4 lg:pt-8'>
+              <DropdownMenu
+                items={[
+                  {
+                    id: "all",
+                    title: "Alle Daten",
+                    onClick: async () => {
+                      const allRecords = await getRecordsBySensorId(sensor.id);
+                      downloadCSV(
+                        createCSVStructure(allRecords),
+                        `${moment.parseZone().format("YYYY-MM-DD")}-sensor-${
+                          sensor.id
+                        }-all-data`
+                      );
+                    },
+                  },
+                  {
+                    id: "all",
```

Comment:
There was an issue with a duplicate key which probably originates here. ID should probably be something like `filtered`